s3transfer/futures.py

```diff
@@ -219,14 +224,30 @@ def cancel(self):
         """Cancels the TransferFuture"""
         if not self.done():
             with self._lock:
```

Comment:
Should the entire method be protected by a lock?  `cancel()` would only be called in error cases anyways, and as is, I think it's still possible to have a race condition where between the if statement and lock acquisition some other thread can come along and call a method that changes the state.